### PR TITLE
add GET /api/articles/:article_id/comments to get comments for a specified article_id

### PR DESCRIPTION
**Pull request by dku7**
Added  endpoint `/api/articles/:article_id/comments` to get all comments for a specified `article_id` sorted by newest comments first.

- Tested for:
  - `GET`: `200` - successfully get all comments for the specified `article_id`
  - `GET`: `404` - the specified `article_id` does not exist
  - `GET`: `400` - the specified `article_id` is not in the expected format

- created new `controller` and `model` functions
- updated `endpoints.json` to include new endpoint

**Note:** 
It is a valid scenario that an `article` has no `comments`. Therefore, the controller function, `getCommentsByArticleId()`, first checks for a valid `article_id` before calling `selectCommentsByArticleId()`. If the `article_id` _is_ invalid, the error is caught and handled by the promise chain.

It is important that this is handled by the controller as the usual check for zero rows within the model function would not make sense in this case, and could incorrectly result in a `404` 'Not found'.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -106,3 +106,40 @@ describe("/api/articles", () => {
...
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: 1,
+          })
+        );
+
+        expect(comments).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+
```
Comment (nc-jodie-cheung) [inline __tests__/app.test.js:131]: You could add in a test for a status 200, valid ID, but has no comments and responds with an empty array of comments 💡

File: models/comments.model.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    WHERE article_id = $1
+    ORDER BY created_at DESC
+    `,
+        [article_id]
+      )
+      // N.B. no need to test for empty rows as it is valid for no comments
+      // to be attached to an article. We can assume that article_id has been
+      // validated by controller before we query the database
+      .then((results) => results.rows)
+  );
```
Comment (nc-jodie-cheung) [inline models/comments.model.js:25]: What happens if there are no comments? Ideally you would want a promise.reject here 💡

File: __tests__/app.test.js (hunk 1 above)
Comment (dku7) [reply __tests__/app.test.js:131]: This was added in PR https://github.com/dku7/be-nc-news/pull/8

File: models/comments.model.js (hunk 2 above)
Comment (dku7) [reply models/comments.model.js:25]: Would a `Promise.reject` not result in an error rather than an empty array when the `article_id` is valid but there are no comments for it?
